src/parse.cpp

```diff
@@ -566,11 +561,14 @@ expression* parser::parse_async_expression_only(token async_token) {
                 .arrow = this->peek().span(),
             });
       }
-      if (await_as_parameter_token) {
-        this->error_reporter_->report(
-            error_await_parameter_disallowed_in_async_arrow_function{
-                .await_token = *await_as_parameter_token,
-            });
+      for (auto parameter : parameters) {
+        if (parameter->variable_identifier_token_type() ==
+            token_type::kw_await) {
```

Comment:
Blocking: What if the parameter isn't a variable? E.g. `([destructured, array]) => {}`. Your code will crash. So I think we need to check `parameter->kind()`.